### PR TITLE
Fix gce util script for getting PREEXISTING_NETWORK_MODE

### LINKED ISSUE #5019: pr-gpu-gce job is leaking custom sub-network
open the issue for tracking

/assign @MrHohn 
cc @mindprince @rmmh 

@MrHohn thinks https://github.com/kubernetes/kubernetes/pull/53751 might break the deletion of subnetwork

We might want to flip back to auto-network until the issue is resolved?

### PR DESCRIPTION
**Pull request by MrHohn**
**What this PR does / why we need it**:
Using `x_gcloud_subnet_mode` seems return nothing and broke the custom network cleanup logic, causing [pull-kubernetes-e2e-gce-gpu](https://k8s-testgrid.appspot.com/kubernetes-presubmits#pull-kubernetes-e2e-gce-gpu) constantly leaking network resource.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes https://github.com/kubernetes/test-infra/issues/5019

**Special notes for your reviewer**:
/assign @bowei 
cc @jingax10 @krzyzacy @mindprince 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by MrHohn** [comment]
Seems like gcloud 174.0.0 has `x_gcloud_mode: auto` while 175.0.0 has `x_gcloud_subnet_mode: auto`...

**Comment by krzyzacy** [comment]
hummmmm, did they deprecate `x_gcloud_mode`?

**Comment by MrHohn** [comment]
> hummmmm, did they deprecate x_gcloud_mode?

Yeah that is what we believe :)

**Comment by MrHohn** [comment]
This PR is not enough, will reopen with more appropriate changes.

**Comment by krzyzacy** [comment]
I can bump gcloud version and see if that fixes things

**Comment by MrHohn** [comment]
> I can bump gcloud version and see if that fixes things

Hoping that will not break more stuffs. I think it will also be nice if we can support both gcloud version in script so folks won't have to upgrade their gcloud locally.

**Comment by MrHohn** [comment]
Most tests passed, PR is ready for review @jingax10 @bowei

**Comment by MrHohn** [comment]
Adding hold so that I will remember to remove the test commit.

/hold

/cc @shyamjvs 
This seems to also affect subnets cleanup in scale CIs, would be great if you could help taking a look :)

Ref https://github.com/kubernetes/kubernetes/issues/52347.

File: cluster/gce/config-test.sh [hunk 1]
```diff
@@ -43,7 +43,7 @@ PREEMPTIBLE_NODE=${PREEMPTIBLE_NODE:-false}
 PREEMPTIBLE_MASTER=${PREEMPTIBLE_MASTER:-false}
 KUBE_DELETE_NODES=${KUBE_DELETE_NODES:-true}
 KUBE_DELETE_NETWORK=${KUBE_DELETE_NETWORK:-true}
-CREATE_CUSTOM_NETWORK=${CREATE_CUSTOM_NETWORK:-false}
+CREATE_CUSTOM_NETWORK=${CREATE_CUSTOM_NETWORK:-true}
```
Comment (shyamjvs) [inline cluster/gce/config-test.sh]: I'm not sure I understand why you're changing this default?

File: cluster/kubemark/gce/config-default.sh [hunk 2]
```diff
@@ -32,7 +32,7 @@ MASTER_ROOT_DISK_SIZE=${KUBEMARK_MASTER_ROOT_DISK_SIZE:-$(get-master-root-disk-s
 REGISTER_MASTER_KUBELET=${REGISTER_MASTER:-false}
 PREEMPTIBLE_NODE=${PREEMPTIBLE_NODE:-false}
 NODE_ACCELERATORS=${NODE_ACCELERATORS:-""}
-CREATE_CUSTOM_NETWORK=${CREATE_CUSTOM_NETWORK:-false}
+CREATE_CUSTOM_NETWORK=${CREATE_CUSTOM_NETWORK:-true}
```
Comment (shyamjvs) [inline cluster/kubemark/gce/config-default.sh]: Also change this in cluster/gce/config-default.sh

File: cluster/gce/config-test.sh (hunk 1 above)
Comment (MrHohn) [reply cluster/gce/config-test.sh]: They are in the `[REMOVE ME]` commit. I added this to prove my changes fixes the issue :)

Comment (shyamjvs) [reply cluster/gce/config-test.sh]: Sorry, missed that.

**Comment by MrHohn** [comment]
@shyamjvs Sorry for the confusion. I removed the `[REMOVE ME]` commit.

/unhold

**Comment by MrHohn** [comment]
/cc @bowei 
for approval.

/priority failing-test

**Pull request merged**